product_pricelist_direct_print/wizards/product_pricelist_print.py

```diff
@@ -294,8 +295,16 @@ def get_products_to_print(self):
                 )
             else:
                 products = self.product_tmpl_ids
-        if not products:
+        domain = self.get_products_domain()
+        # We default to the products model search but if the last ordered products
+        # checkbox is on we don't want to confuse the user with products which don't
+        # belong to their sales. It's better to show the empty report in that case.
+        if not products and not self.last_ordered_products:
             products = products.search(self.get_products_domain())
+        else:
+            # We need to filter the results anyway, as there could be some extra filters
+            # to consider, like categories.
+            products = products.filtered_domain(domain)
```

Comment:
I'm in doubt here. Should we filter manually selected products?

Reply:
If the products are selected manually is because they want to print each products, so I think that it shouldn't be filtered

Reply:
Yes, I feel the same. It's a matter of UI then.